kroxylicious/src/main/java/io/kroxylicious/proxy/internal/codec/KafkaRequestDecoder.java

```diff
@@ -85,12 +89,19 @@ protected Frame decodeHeaderAndBody(ChannelHandlerContext ctx, ByteBuf in, final
 
         RequestHeaderData header = null;
         final ByteBufAccessorImpl accessor;
+        Metrics.globalRegistry.counter("inbound_downstream_messages").increment();
         var decodeRequest = decodePredicate.shouldDecodeRequest(apiKey, apiVersion);
         LOGGER.debug("Decode {}/v{} request? {}, Predicate {} ", apiKey, apiVersion, decodeRequest, decodePredicate);
         boolean decodeResponse = decodePredicate.shouldDecodeResponse(apiKey, apiVersion);
         LOGGER.debug("Decode {}/v{} response? {}, Predicate {}", apiKey, apiVersion, decodeResponse, decodePredicate);
         short headerVersion = apiKey.requestHeaderVersion(apiVersion);
         if (decodeRequest) {
+            Metrics.globalRegistry.counter("decoded_messages").increment();
```

Comment:
Probably should include the `inbound_downstream` prefix here as we decode on the upstream side to.

Thinking out loud, would it make more sense for it to be a single `decoded_messages` counter with `direction` and `flowing` tags for example `counter("decoded_messages, "direction", "inbound", "flowing", "downstream")`? 

cc: @robobario thoughts?

Reply:
what's the difference between direction and flowing? Definitely keen to use one counter with tags rather than prefixes.

Reply:
Also by convention prometheus recommends `_total` suffixes for counters. I think micrometer might add that suffix automatically but it's nice to have it explicit in code for grepping.

Reply:
keen to introduce at least one level of prefixing though like `kroxylicious_`

Reply:
After a chat:

We would potentially like to be able to measure metrics like:

1. bytes/messages from client to proxy (direction: in)
2. bytes/messages from proxy to broker (direction: in)
3. bytes/messages from browser to proxy (direction: out)
4. bytes/messages from proxy to client (direction: out)

Maybe we could use two sets of metrics that describe what kroxy is doing:

`kroxylicious_bytes_received_total{direction=in}`
`kroxylicious_bytes_sent_total{direction=in}`
`kroxylicious_bytes_received_total{direction=out}`
`kroxylicious_bytes_sent_total{direction=out}`

Reply:
> Maybe we could use two sets of metrics that describe what kroxy is doing:
> 
> `kroxylicious_received_bytes_total{direction=in}` `kroxylicious_sent_bytes_total{direction=in}` `kroxylicious_received_bytes_total{direction=out}` `kroxylicious_sent_bytes_total{direction=out}`
I was actually thinking of:

- `kroxylicious_received_bytes_total{flowing=upstream}` from client to proxy traffic
- `kroxylicious_sent_bytes_total{flowing=upstream}` from proxy to cluster traffic 
- `kroxylicious_received_bytes_total{flowing=downstream}` from cluster to proxy traffic
- `kroxylicious_sent_bytes_total{flowing=downstream}` from proxy to client traffic 

So using the `flow` terminology` inline with the [defined terms](https://github.com/kroxylicious/design/blob/main/concepts.asciidoc#upstream-vs-downstream) in the design repo.

Reply:
Cool, it says:
> It will later on be merged into some more formal kind of documentation probably.

As soon as we are using the term in metrics sounds like a good point to put it in docs somewhere